cmd/setec/setec.go

```diff
@@ -140,3 +211,151 @@ func runServer(ctx context.Context, args []string) error {
 
 	return nil
 }
+
+func clientFromArgs(args []string) (client *setec.Client, remainingArgs []string, err error) {
+	if len(args) < 1 {
+		return nil, nil, flag.ErrHelp
+	}
+	server := args[0]
+	// TODO(corp/13375): make a better UX here where single-label
+	// hostnames get expanded into a full HTTPS URL.
+	return &setec.Client{
+		Server: server,
+	}, args[1:], nil
+}
+
+func runList(ctx context.Context, args []string) error {
+	c, args, err := clientFromArgs(args)
+	if err != nil {
+		return err
+	}
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	secrets, err := c.List(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to list secrets: %v", err)
+	}
+
+	tw := tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0)
+	io.WriteString(tw, "NAME\tACTIVE\tVERSIONS\n")
+	for _, s := range secrets {
+		vers := make([]string, 0, len(s.Versions))
+		for _, v := range s.Versions {
+			vers = append(vers, v.String())
+		}
+		fmt.Fprintf(tw, "%s\t%s\t%s\n", s.Name, s.ActiveVersion, strings.Join(vers, ","))
+	}
+	return tw.Flush()
+}
+
+var getArgs struct {
+	Version uint64
+}
+
+func runGet(ctx context.Context, args []string) error {
+	c, args, err := clientFromArgs(args)
+	if err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return flag.ErrHelp
+	}
+
+	name := args[0]
+
+	var val *api.SecretValue
+	if getArgs.Version == 0 {
+		val, err = c.Get(ctx, name)
+	} else {
+		val, err = c.GetVersion(ctx, name, api.SecretVersion(getArgs.Version))
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get secret: %v", err)
+	}
+
+	// Print with a newline if a human's going to look at it,
```

Comment:
Although at the moment these are mostly printable text since we're debugging, I expect a lot of them in production will be binary goop. Perhaps we can (later) do something like:

```go
if !term.IsTerminal(int(os.Stdout.Fd())) {
   os.Stdout.Write(val.Value)
} else if utf8.Valid(val.Value) {
   fmt.Println(string(val.Value))
} else {
   fmt.Printf("%x\n", val.Value)
}
```

Not a big deal, though.